app/application.go

```diff
@@ -27,21 +28,30 @@ func NewApplication(ctx context.Context) Application {
 // bootstrap
 
 func (a *application) Setup() {
-	//viper
 	config, err := utils.SetupViper("config/config.yml")
 	a.config = config
 
 	if err != nil {
 		log.Fatalf("failed to setup viper: %s", err.Error())
 	}
+
 	app := fx.New(
 		fx.Provide(
+			a.InitDatabase,
```

Comment:
here we have to change the order of injection. _DI is done from bottom to top_
Because our `repository` needs a `Database`, we must set up the `Database` at a lower layer than the `repository`.

```
		fx.Provide(
			a.InitRouter,
			a.InitFramework,
			a.InitController,
			a.InitServices,
			a.InitRepositories,
			**a.InitDatabase**,
		),
```